mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java

```diff
@@ -110,6 +114,27 @@ public Mailbox findMailboxByPath(MailboxPath path) throws MailboxException {
         }
     }
 
+    private CompletableFuture<Optional<SimpleMailbox>> fromPreviousTable(MailboxPath path) throws MailboxException {
+        try {
+            return mailboxPathDAO.retrieveId(path)
+                .thenCompose(cassandraIdOptional ->
+                    cassandraIdOptional
+                        .map(CassandraIdAndPath::getCassandraId)
+                        .map(this::retrieveMailbox)
+                        .orElse(CompletableFuture.completedFuture(Optional.empty())));
+        } catch (CompletionException e) {
+            if (e.getCause() instanceof InvalidQueryException) {
+                String errorMessage = e.getCause().getMessage();
+                if (StringUtils.containsIgnoreCase(errorMessage, VALUES_MAY_NOT_BE_LARGER_THAN_64_K) 
```

Comment:
This `if condition` needs to be extracted.

I would value having a `DriverExceptionHelper` class in the backend managing it ^^